omnibus/config/software/datadog-agent.rb

```diff
@@ -237,4 +240,35 @@
   unless windows_target?
     delete "#{install_dir}/uselessfile"
   end
+
+  # TODO: move this to omnibus-ruby::health-check.rb
+  # check that linux binaries contains OpenSSL symbols when building to support FIPS
+  if fips_mode? && linux_target?
+
+      # Put the ruby code in a block to prevent omnibus from running it directly but rather at build step with the rest of the code above.
+      # If not in a block, it will search for binaries that have not been built yet.
+      block do
+        LINUX_BINARIES = [
+          "#{install_dir}/bin/agent/agent",
+          "#{install_dir}/embedded/bin/trace-agent",
+          "#{install_dir}/embedded/bin/process-agent",
+          "#{install_dir}/embedded/bin/security-agent",
+          "#{install_dir}/embedded/bin/system-probe",
+        ]
+
+        symbol = "_Cfunc_go_openssl"
+        check_block = Proc.new { |symbols|
+          count = symbols.scan(symbol).count
+          if count > 0
+            log.info(log_key) { "Symbol '#{symbol}' found #{count} times in binary '#{@binary}'." }
```

Comment:
Same comment for the `log_key` as this one https://github.com/DataDog/datadog-agent/pull/29560/files#r1778385309